src/dune_rules/include_preprocessed_sources.ml

```diff
@@ -0,0 +1,265 @@
+                      external source *)
+                   let info = Lib.info lib in
+                   let src_dir = Lib_info.src_dir info in
+                   if Path.is_descendant ~of_:(Path.build dir) src_dir
+                   then
+                     let+ src_to_pp =
+                       Lib_or_exes_to_pp.from_lib ~sctx ~scope ~expander lib
+                     in
+                     append_opt src_to_pp acc
+                   else Memo.return acc)
+              | Dune_file.Executables exes ->
+                let+ src_to_pp =
+                  Lib_or_exes_to_pp.from_exes ~sctx ~scope ~expander ~dir exes
+                in
+                append_opt src_to_pp acc
+              | _ -> Memo.return acc)))
+  in
+  Appendable_list.to_list srcs_to_pp
+;;
+
+let gen_rule_for_source_file ~sctx ~dir ~ppx_driver_and_flags ~ml_kind path =
+  let target = Path.Build.append_source dir (Path.drop_build_context_exn path) in
+  (*let target = Path.Build.relative dir (Path.basename path) in*)
+  let rule =
+    Action_builder.with_file_targets
+      ~file_targets:[ target ]
+      (let open Action_builder.O in
+       let* ppx_driver, flags = ppx_driver_and_flags in
+       Command.run'
+         ~dir:(Path.build dir)
+         (Ok (Path.build ppx_driver))
+         [ As flags
+         ; Command.Ml_kind.ppx_driver_flag ml_kind
+         ; Dep path
+         ; A "-o"
+         ; Path (Path.build target)
+         ])
+  in
+  let mode = Rule.Mode.Promote { lifetime = Unlimited; into = None; only = None } in
+  Super_context.add_rule sctx ~mode ~dir rule
+;;
+
+let gen_rules_for_module ~sctx ~dir ~ppx_driver_and_flags module_ =
+  let impl = Module.file ~ml_kind:Impl module_ in
+  let intf = Module.file ~ml_kind:Intf module_ in
+  let* () =
+    match impl with
+    | None -> Memo.return ()
+    | Some path ->
+      gen_rule_for_source_file ~sctx ~dir ~ppx_driver_and_flags ~ml_kind:Impl path
+  in
+  match intf with
+  | None -> Memo.return ()
+  | Some path ->
+    gen_rule_for_source_file ~sctx ~dir ~ppx_driver_and_flags ~ml_kind:Intf path
+;;
+
+let gen_rules_for_lib_or_exes ~sctx ~dir lib_or_exes_to_pp =
+  let { Lib_or_exes_to_pp.sources; ppx_driver_and_flags } = lib_or_exes_to_pp in
+  Memo.List.iter sources ~f:(gen_rules_for_module ~sctx ~dir ~ppx_driver_and_flags)
+;;
+
+let gen_stanza_rules ~dir ~dirs_to_exclude sctx =
+  let* scope = Scope.DB.find_by_dir dir in
+  let* expander = Super_context.expander sctx ~dir in
+  let dirs_to_exclude = List.map dirs_to_exclude ~f:Path.drop_build_context_exn in
+  let* libs_or_exes_to_pp =
+    libs_or_exes_to_pp_in_source_tree ~sctx ~scope ~expander ~dirs_to_exclude
+  in
+  Memo.List.iter libs_or_exes_to_pp ~f:(gen_rules_for_lib_or_exes ~sctx ~dir)
+;;
+
+type t = { dirs_to_exclude : String_with_vars.t list }
+type Stanza.t += T of t
+
+let syntax =
+  let name = "include_preprocessed_sources" in
+  let desc = "dummy" in
+  Dune_lang.Syntax.create ~name ~desc [ (0, 1), `Since (3, 10) ]
```

Comment:
We now have 3.13 in main if you want to use that.

Reply:
Awesome, I'll switch to it! I rolled it back to 3.10 while trying to fix an issue I had with merlin locally.

Reply:
I did bump it to 3.13 but since the PR also enables the extension it makes the whole repo depend on `dune >= 3.13` and I therefore get CI builds failure with:
```
Switch invariant: ["ocaml" {>= "4.05.0"}]
[ERROR] Could not determine which packages to install for this switch:
  * Missing dependency:
Switch initialisation failed: clean up? ('n' will leave the switch partially installed) [Y/n] y
    - dune >= 3.13
    no matching version
 ```
 This is expected I guess. Is there any way I can tell the CI builds to pin `dune.3.13.0` instead of `3.11.1`?

Reply:
Maybe its best to keep it 3.11 then. I don't think its worth spending time resolving these CI issues. It's not a user facing feature anyway yet.

Reply:
To be clear, the issue is not because we are adding a field to a new dune lang version, but because the CI works by coincidence. The actual dune-project version Dune itself uses, always lags behind the latest dune language version.

Reply:
I could also not enable it in this PR, simply adding the feature and enabling it later if that helps.

I'm also happy to use an older version of the language as lower bound for the feature since, as you said, it's not user facing.